runtime/drivers/duckdb/duckdb.go

```diff
@@ -409,3 +439,126 @@ func (c *connection) checkErr(err error) error {
 	}
 	return err
 }
+
+// Periodically collects stats using pragma_database_size() and emits as activity events
+func (c *connection) periodicallyEmitStats(d time.Duration) {
+	if c.config.Activity == nil {
+		// Activity client isn't set, there is no need to report stats
+		return
+	}
+
+	statTicker := time.NewTicker(d)
+	for {
+		select {
+		case <-statTicker.C:
+			var stat dbStat
+			// Obtain a connection, query, release
+			err := func() error {
+				conn, release, err := c.acquireMetaConn(c.ctx)
+				if err != nil {
+					return err
+				}
+				defer func() { _ = release() }()
+				err = conn.GetContext(c.ctx, &stat, "CALL pragma_database_size()")
+				return err
+			}()
+			if err != nil {
+				c.logger.Error("couldn't query DuckDB stats", zap.Error(err))
+				continue
+			}
+
+			// Emit collected stats as activity events
+			commonDims := []attribute.KeyValue{
+				attribute.String("olap.db.name", stat.DatabaseName),
+			}
+			commonDims = append(commonDims, c.config.ActivityDims...)
+
+			dbSize, err := humanReadableSizeToBytes(stat.DatabaseSize)
+			if err != nil {
+				c.logger.Error("couldn't convert duckdb size to bytes", zap.Error(err))
+			} else {
+				c.config.Activity.Emit(c.ctx, "olap_db_size_bytes", dbSize, commonDims...)
+			}
+
+			walSize, err := humanReadableSizeToBytes(stat.WalSize)
+			if err != nil {
+				c.logger.Error("couldn't convert duckdb wal size to bytes", zap.Error(err))
+			} else {
+				c.config.Activity.Emit(c.ctx, "olap_wal_size_bytes", walSize, commonDims...)
+			}
+
+			memoryUsage, err := humanReadableSizeToBytes(stat.MemoryUsage)
+			if err != nil {
+				c.logger.Error("couldn't convert duckdb memory usage to bytes", zap.Error(err))
+			} else {
+				c.config.Activity.Emit(c.ctx, "olap_memory_usage_bytes", memoryUsage, commonDims...)
+			}
+
+			memoryLimit, err := humanReadableSizeToBytes(stat.MemoryLimit)
+			if err != nil {
+				c.logger.Error("couldn't convert duckdb memory limit to bytes", zap.Error(err))
+			} else {
+				c.config.Activity.Emit(c.ctx, "olap_memory_limit_bytes", memoryLimit, commonDims...)
+			}
+
+			c.config.Activity.Emit(c.ctx, "olap_block_size_bytes", float64(stat.BlockSize), commonDims...)
+			c.config.Activity.Emit(c.ctx, "olap_total_blocks", float64(stat.TotalBlocks), commonDims...)
+			c.config.Activity.Emit(c.ctx, "olap_free_blocks", float64(stat.FreeBlocks), commonDims...)
+			c.config.Activity.Emit(c.ctx, "olap_used_blocks", float64(stat.UsedBlocks), commonDims...)
```

Comment:
Since these metrics are very duckdb-specific, consider changing the `olap_` prefix to `duckdb_`